src/operators.rs

```diff
@@ -212,6 +212,51 @@ where
     }
 }
 
+// Matrix operations
+
+/// Implements matrix multiplication for Dimensional arrays.
+impl<T: Num + Copy, S> Dimensional<T, S, 2>
+where
+    S: DimensionalStorage<T, 2>,
+{
+    /// Multiplies two matrices.
+    pub fn mat_mul(&self, rhs: &Self) -> Self {
+        assert_eq!(
+            self.shape()[1], rhs.shape()[0],
+            "Matrix dimensions must match for multiplication"
+        );
+        let (rows, cols) = (self.shape()[0], rhs.shape()[1]);
+
+        Self::from_fn([rows, cols], |[i, j]| {
+            (0..self.shape()[1]).fold(T::zero(), |sum, k| sum + self[[i, k]] * rhs[[k, j]])
+        })
+    }
+}
+
+/// Implements matrix transpose for Dimensional arrays.
+impl<T: Num + Copy, S> Dimensional<T, S, 2>
+where
+    S: DimensionalStorage<T, 2>,
+{
+    /// Transposes a matrix.
+    pub fn transpose(&self) -> Self {
```

Comment:
Do you think there is a zero copy way to do this? Maybe a todo for later.

Reply:
Tried a bit but it got a bit verbose, adding as todo for now https://github.com/warlock-labs/dimensionals/pull/4/commits/3df8ee944d2de78c37f07fd5c83c7f0a5a617d00

Reply:
Nice.